src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2966,6 +2974,16 @@ def aks_update(cmd, client, resource_group_name, name,
             tier="Free"
         )
 
+    if disable_local_accounts and enable_local_accounts:
+        raise CLIError('Cannot specify --disable-local-accounts and --enable-local-accounts '
+                       'at the same time.')
+
+    if disable_local_accounts:
+        instance.disable_local_accounts = True
+
+    if enable_local_accounts:
+        instance.disable_local_accounts = False
```

Comment:
Should only one parameter meet the requirements? Such as:

```
if disable_local_accounts != None:
     instance.disable_local_accounts = disable_local_accounts
```

Reply:
Two points:
1. AKS's command has many other similar settings, such as enable/disable public fqdn, enable/disable ahub, etc. This feature uses existing pattern.
2. If I understand it correctly, for `store_true` option, it will be either true or false, and will never be None, default value will be false. In this case, customer must explicitly specify this parameter every time, otherwise their setting can be changed unexpectedly: current `disableLocalAccounts` is true, and if they don't explicitly specify `--disable-local-accounts` to false every time they update the AKS cluster, the setting will be updated to false.

Reply:
I mean, if you keep only one parameter `disable_local_accounts`, the code here can be simplified to `instance.disable_local_accounts = disable_local_accounts`.
At present, it is designed as two parameters with opposite meanings to do the same thing, which is a little redundant. What do you think of it? @FumingZhang

Reply:
What if customer does not provide this option? always use the default value `false`?

Reply:
I guess only one option is not enough for the user to set this property. For option `--disable-local-accounts`, its default value is False. In the **update** command, if the user does not set this option **explicitly** (that is, the value of `disable_local_accounts` is still False), I suppose it does not mean that the user wants to enable local accounts (that is, set `instance.disable_local_accounts = False`), but maintains the original setting.

Reply:
Could we not update this property when users does not pass in this parameter `--disable-local-accounts`?

Reply:
How to determine whether user passes this property? The `store_true` option will be set to default value `False` if user does not pass the property.

Reply:
> What if customer does not provide this option? always use the default value false?

@norshtein When using `store_true`, if users do not input this parameter, it's value is `None`. If this parameter is inputed, it's value becomes `True`.

Reply:
> > What if customer does not provide this option? always use the default value false?
> 
> @norshtein When using `store_true`, if users do not input this parameter, it's value is `None`. If this parameter is inputed, it's value becomes `True`.

Are you sure about it? The doc says the default value will be `False` https://docs.python.org/3/library/argparse.html#action

Reply:
@FumingZhang  I don't think `store_true` is suitable for this scenario and `get_three_state_flag()` may be a better choice.
In this way, the users can explicitly pass in `True` or `False` to enable or disable the property. When users do not pass in this parameter, the value will be `None`, then the property should not be updated

Reply:
Several reasons to keep current option unchanged:
1. The option has been in aks-preview for months, if we change it, when cx switch from aks-preview to Azure CLI, it will be a breaking change, we should avoid it.
2. Semantically, `--enable-xxx` and `disable-xxx` is more clear to cx. `disable-xxx=True` and `disable-xxx=False` can cause confusion to cx.
3. AKS update command has many existing options using `--enable-xxx` and `--disable-xxx`, why only this one is blocked? https://github.com/Azure/azure-cli/pull/19302

Reply:
> Are you sure about it? The doc says the default value will be False https://docs.python.org/3/library/argparse.html#action

Yes, I confirmed this through code debug.

Reply:
> 1. The option has been in aks-preview for months, if we change it, when cx switch from aks-preview to Azure CLI, it will be a breaking change, we should avoid it.

OK, I agree with this point

> 2. Semantically, --enable-xxx and disable-xxx is more clear to cx. disable-xxx=True and disable-xxx=False can cause confusion to cx.

The commands of other modules always use only one parameter, such as `--disable-local-auth`, `--enable-session` and so on. So the CLI users are used to this approach. I personally think that two parameters doing the same thing will appear redundant, and the parameters become more and more difficult to find.

> 3. AKS update command has many existing options using --enable-xxx and --disable-xxx, why only this one is blocked?

My point is not only for this parameter, but for all parameters used in this way. I'm sorry I didn't notice this detail before

Reply:
Because this is to avoid breaking change and keep consistent with other parameter design style of the AKS module, so I think this design is acceptable